### PR TITLE
Update to 9.0.100-rc.1

### LINKED ISSUE #2789: Update to 9.0.100-rc1
Breaking changes due to https://learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib0057

Several places need to be fixed, especially in the tests. right now it's set to warn in the common.props
https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/dev/build/common.props#L48

and here:
https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/dev/build/commonTest.props#L32

Todo:
remove the two <warn> in the props and address all the issues.
Will probably need
`#if NET9_0_OR_GREATER`
in all the places

### PR DESCRIPTION
**Pull request by kllysng**
Fixes #2789

### CODE REVIEW

File: build/template-Build-run-tests-sign.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ steps:
 - task: UseDotNet@2
   displayName: 'Use .Net Core SDK 9.x'
   inputs:
-    version: 9.0.100-preview.4.24267.66
+    version: 9.0.100-preview.7.24407.12
```
Comment (jennyf19) [inline build/template-Build-run-tests-sign.yml]: we should move to RC1 https://dotnet.microsoft.com/en-us/download/dotnet/9.0

Comment (pmaytak) [inline build/template-Build-run-tests-sign.yml]:
```suggestion
    version: 9.0.0-rc.1.24431.7
```

Comment (jennyf19) [reply build/template-Build-run-tests-sign.yml]: would like @kellyyangsong to take this locally and make sure the tests work locally instead of taking the suggestion via the UI.

File: src/Microsoft.IdentityModel.Protocols.WsFederation/Configuration/WsFederationConfigurationValidator.cs [hunk 2]
```diff
@@ -147,7 +147,11 @@ public ConfigurationValidationResult Validate(WsFederationConfiguration configur
 
                     try
                     {
+#if NET9_0_OR_GREATER
+                        cert = X509CertificateLoader.LoadCertificate(Convert.FromBase64String(signatureCertData.Current));
+#else
                         cert = new X509Certificate2(Convert.FromBase64String(signatureCertData.Current));
+#endif
```
Comment (pmaytak) [inline src/Microsoft.IdentityModel.Protocols.WsFederation/Configuration/WsFederationConfigurationValidator.cs]: Since these lines repeat throughout, one code duplication reduction to consider is extract this into a helper method, so that `#if#` flag is only in one place.

```csharp
cert = CertificateHelper.Load(Convert.FromBase64String(signatureCertData.Current));
```
```csharp
internal static X509Certificate2 Load(byte[] data)
{
#if NET9_0_OR_GREATER
    return X509CertificateLoader.LoadCertificate(data);
#else
    return new X509Certificate2(data);
#endif
}
```

File: test/Microsoft.IdentityModel.TestUtils/KeyingMaterial.cs [hunk 3]
```diff
@@ -71,9 +85,15 @@ static public class KeyingMaterial
         public static SigningCredentials X509SigningCreds_SelfSigned2048_SHA512 = new SigningCredentials(X509SecurityKeySelfSigned2048_SHA512, SecurityAlgorithms.RsaSha512);
 
         // all asymmetric material has private key unless public is included in variable name
-        public static SecureString CertPassword = ConvertToSecureString("abcd");
+        public static string CertPasswordString = "abcd";
+        public static SecureString CertPassword = ConvertToSecureString(CertPasswordString);
```
Comment (pmaytak) [inline test/Microsoft.IdentityModel.TestUtils/KeyingMaterial.cs]: Why was this change and the ones above made?

**Review by pmaytak — COMMENTED** [review]
Main suggestion is to refactor the certificate creation into a separate helper. This will also help, for example, if in the future we decide to use `Microsoft.Bcl.Cryptography` package, then the code change will only have to be in one place.

File: build/template-Build-run-tests-sign.yml (hunk 1 above)
Comment (pmaytak) [reply build/template-Build-run-tests-sign.yml]: Whatever works best.

File: test/Microsoft.IdentityModel.TestUtils/KeyingMaterial.cs (hunk 3 above)
Comment (kllysng) [reply test/Microsoft.IdentityModel.TestUtils/KeyingMaterial.cs]: Ah they're vestigial to before we decided to pragma warning disable since the constructor didn't have an overload. X509CertificateLoader.LoadPcks12() has similar args but with a string instead of secure string password arg. But we won't be using that - so reverting these changes. Thank you!

File: src/Microsoft.IdentityModel.Protocols.WsFederation/Configuration/WsFederationConfigurationValidator.cs (hunk 2 above)
Comment (kllysng) [reply src/Microsoft.IdentityModel.Protocols.WsFederation/Configuration/WsFederationConfigurationValidator.cs]: Great point, added a `X509CertificateHelper`.

File: build/template-Build-run-tests-sign.yml (hunk 1 above)
Comment (kllysng) [reply build/template-Build-run-tests-sign.yml]: Updated to rc1.

Comment (kllysng) [reply build/template-Build-run-tests-sign.yml]: Updated to rc1.

File: test/Microsoft.IdentityModel.TestUtils/KeyingMaterial.cs (hunk 3 above)
Comment (kllysng) [reply test/Microsoft.IdentityModel.TestUtils/KeyingMaterial.cs]: Reverted these changes.

File: test/Microsoft.IdentityModel.Xml.Tests/EnvelopedSignatureReaderTests.cs [hunk 4]
```diff
@@ -213,8 +212,8 @@ public static TheoryData<EnvelopedSignatureTheoryData> ReadXmlElementsTheoryData
...
...tk2bXMYq8dO3xSRASmdVmgmoyoQGJrnh1E0SihxCFNgiHqyGQ1JAgMBAAGjggE9MIIBOTAJBgNVHRMEAjAAMBEGCWCGSAGG+EIBAQQEAwIEsDAdBgNVHQ4EFgQU0HAvIqGfYUW/akxHd9SNK2O4GwEwgbsGA1UdIwSBszCBsIAUzxJQXpMfXwleF48WV0F6n3lyhLihgYykgYkwgYYxCzAJBgNVBAYTAkNaMQ8wDQYDVQQHEwZQcmFndWUxGTAXBgNVBAoTEENaLk5JQywgei5zLnAuby4xMTAvBgNVBAMTKENaLk5JQyBTSEEyIFJvb3QgQ2VydGlmaWNhdGlvbiBBdXRob3JpdHkxGDAWBgkqhkiG9w0BCQEWCWNhQG5pYy5jeoIJAJxtRGsvNinfMBQGA1UdEgQNMAuBCWNhQG5pYy5jejAWBgNVHREEDzANgQtyb290QG5pYy5jejAOBgNVHQ8BAf8EBAMCBaAwDQYJKoZIhvcNAQELBQADggIBAIytUqXh7AU/OunkSpUTaEY5Ze2sdV76JPwYVzNn2O6hCyzGvPXARP0IchUv8Vy+eCFLctUxvCbUS6aC+ObKgGXq4MxoSV8lijMLEW9crCpFDoLLd3LQw0GjVk2mCE7XTuaT0choPYlZmjv+wF2ZKm5/B+Qjek2j7SkY1yn7hxgJdd5ljHE6wmDXLJ8gHuVBNwvc5iHDjHYh7jL5c5jCDBcr1fFCsIARU05RAkfpWurl8GKY8t6IPm7iopOLjru3Gl45ZBdVrAPMQ8Fz1M9VElUJ6ngeKkXHkSwGhCBG3X0MYsltND6mZSkJqN0nOs+cJ6HHO8IZW1f2pH3aCTUSDYWoaZbK74NC6d61sr5Rth4foLQnMzCS5RaXuANMvyZW3Ol5ScvLl/KRZM4f4CB6rmYinyHfXIoPF+uCjavyOYnW1RDBASg0Ld/WUJlWb75m5GNkRELIc4c5FU54ysMW9o5wnGpQvtXNdCBJK8tAyZO9Wf2hjZeOZgJ6r1IngfSeSFu7EOFqWnwVOF+3juwWOLCwxrKcURAEngwH01ydwU3oG/rN+7JtdS3IwfaBt9sfDiLQ60qec/6PQc643UztE6oToHLRXsidwrObwyAKLSFJoh/uxWT85JgAoekq5zBen94HfELfMEc9tex6Qlf1tLDs7OWD6Mlw6j9aAcw/4Nfh")));
```
Comment (kllysng) [inline test/Microsoft.IdentityModel.Xml.Tests/EnvelopedSignatureReaderTests.cs]: Had to add a `X509CertificateHelper` to this package as adding an internals visible to for `Microsoft.IdentityModel.Tokens.Xml.Tests` to the Wilson.Tokens package assembly info didn't work like it did for Wilson.Protocols.WsFederation, for example. The only thing I can think of is because the namespace of this file doesn't match the package/.csproj, which is `Microsoft.IdentityModel.Xml.Tests` and not `Microsoft.IdentityModel.Tokens.Xml.Tests`.

Comment (kllysng) [reply test/Microsoft.IdentityModel.Xml.Tests/EnvelopedSignatureReaderTests.cs]: image
Despite adding to Wilson.Tokens assembly info, adding Wilson.Tokens package ref to .csproj, using Wilson.Tokens at the top of this file..

File: src/Microsoft.IdentityModel.Tokens/X509CertificateHelper.cs [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+using System.Security.Cryptography.X509Certificates;
+
+namespace Microsoft.IdentityModel.Tokens
+{
+    /// <summary>
+    /// Helper class to load X509Certificate2 from byte array.
+    /// </summary>
+    internal class X509CertificateHelper
+    {
+        public static X509Certificate2 Load(byte[] data)
```
Comment (pmaytak) [inline src/Microsoft.IdentityModel.Tokens/X509CertificateHelper.cs]: nit: What about naming class more generic like `CertificateHelper` and method `LoadX509Certificate`? That way we can add more methods here to load different cert types in the future, if needed (otherwise, we'd have to create a separate class for each cert type). And since this is an internal class that's used in multiple projects, we should make sure the API is right, since we won't be able to change it.

Comment (pmaytak) [reply src/Microsoft.IdentityModel.Tokens/X509CertificateHelper.cs]: @keegan-caruso Thoughts on adding an internal class to be used in multiple projects? I suppose since we'll add the public API analyzers, this is okay. You mentioned previously an approach of create a separate duplicate helper class in each project to avoid the issues with internal friendlies.

File: test/Microsoft.IdentityModel.Xml.Tests/EnvelopedSignatureReaderTests.cs (hunk 4 above)
Comment (jennyf19) [reply test/Microsoft.IdentityModel.Xml.Tests/EnvelopedSignatureReaderTests.cs]: Not sure I'm following, so the internalsVisibleTo did not work?

File: src/Microsoft.IdentityModel.Tokens/X509CertificateHelper.cs (hunk 5 above)
Comment (kllysng) [reply src/Microsoft.IdentityModel.Tokens/X509CertificateHelper.cs]: I was considering making a separate helper class per project too.. happy to do that. Also happy to name it a bit more generic + .LoadX509()

File: test/Microsoft.IdentityModel.Xml.Tests/EnvelopedSignatureReaderTests.cs (hunk 4 above)
Comment (kllysng) [reply test/Microsoft.IdentityModel.Xml.Tests/EnvelopedSignatureReaderTests.cs]: Did not work, some of the tests in this project have a different namespace than the project namespace.

File: src/Microsoft.IdentityModel.Tokens/X509CertificateHelper.cs (hunk 5 above)
Comment (brentschmaltz) [reply src/Microsoft.IdentityModel.Tokens/X509CertificateHelper.cs]: I agree with @pmaytak a more general name may be an advantage.
Is there a reason why this is internal?

File: src/Microsoft.IdentityModel.Tokens/X509CertificateHelper.cs [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+        {
+#if NET9_0_OR_GREATER
```
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Tokens/X509CertificateHelper.cs]: This API is available in net8 are we choosing net9 deliberately?

File: src/Microsoft.IdentityModel.Tokens/X509CertificateHelper.cs (hunk 5 above)
Comment (kllysng) [reply src/Microsoft.IdentityModel.Tokens/X509CertificateHelper.cs]: Spoke with @keegan-caruso - we'll use [compile include ](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/f0d09d408948c6920f171299bee9167d0850a51b/src/Microsoft.IdentityModel.Logging/Microsoft.IdentityModel.Logging.csproj#L20C1-L22C15) instead of internals visible to. This way we avoid the issues with internals but we keep the helper class in one spot for maintainability.

@brentschmaltz we want to keep this internal. This is a helper class that we don't want to ship.

File: src/Microsoft.IdentityModel.Tokens/X509CertificateHelper.cs (hunk 6 above)
Comment (kllysng) [reply src/Microsoft.IdentityModel.Tokens/X509CertificateHelper.cs]: The breaking change was introduced in net9

File: test/Microsoft.IdentityModel.Xml.Tests/EnvelopedSignatureReaderTests.cs (hunk 4 above)
Comment (kllysng) [reply test/Microsoft.IdentityModel.Xml.Tests/EnvelopedSignatureReaderTests.cs]: I'll be adding a public helper to the tests class to avoid sharing internals and this issue will be fixed

File: src/Microsoft.IdentityModel.Tokens/X509CertificateHelper.cs (hunk 5 above)
Comment (kllysng) [reply src/Microsoft.IdentityModel.Tokens/X509CertificateHelper.cs]: Done

File: test/Microsoft.IdentityModel.Protocols.WsFederation.Tests/WsFederationMessageTests.cs [hunk 7]
```diff
@@ -139,11 +139,12 @@ public static TheoryData<WsFederationSigninMessageTheoryData> WaSignInTheoryData
...
+                X509Certificate2 cert = TestUtils.CertificateHelper.LoadX509Certificate(Convert.FromBase64String(_x509DataADFS));
```
Comment (msbw2) [inline test/Microsoft.IdentityModel.Protocols.WsFederation.Tests/WsFederationMessageTests.cs]: NIT: Why was this moved up here?

File: test/Microsoft.IdentityModel.Protocols.WsFederation.Tests/WsFederationMessageTests.cs [hunk 8]
```diff
@@ -139,11 +139,12 @@ public static TheoryData<WsFederationSigninMessageTheoryData> WaSignInTheoryData
 
                 // customer data
                 string _x509DataADFS = "MIIDCDCCAfCgAwIBAgIQNz4YVbYAIJVFCc47HFD3RzANBgkqhkiG9w0BAQsFADBAMT4wPAYDVQQDEzVBREZTIFNpZ25pbmcgLSBzdHMuc3ViMi5mcmFjYXMzNjUubXNmdG9ubGluZXJlcHJvLmNvbTAeFw0xNTAzMzExMDQyMTNaFw0zNDA1MzAxMDQyMTNaMEAxPjA8BgNVBAMTNUFERlMgU2lnbmluZyAtIHN0cy5zdWIyLmZyYWNhczM2NS5tc2Z0b25saW5lcmVwcm8uY29tMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAxdjzB+wGV6hYekOvWwKoL/DFNBiLQsLx6w02FzcFnpGwR38gVTn/glg9CNSsOT0riRM3/MwU8o2fwseQyVtv9Kee/yvia8cB6GD0CARlYizb6GkJJzMvWkPSas1zpn10Bs3SBBgn0pvAKZCWWir5WJ7DRY32X2yo2do8mQftsoLGYsEU8+jj9AMYQWaR3A86AEWjXoQY3AodfMMzdVFX+O/kjsvKcBfPqGRT6jUSGBOOaqzMOJBT39SueD8zePDW7SejBl7fRi4TLx5H6xuMldOAAH6oD70yIrobqosGG9X/LdijHajMSoaYzZIlG7fl4PCVvAjh1Dytw/y8K70flQIDAQABMA0GCSqGSIb3DQEBCwUAA4IBAQBy08dAsSTKd2YlNE8bM4O5C2bFKR1YakR8L/zLEy8g+RNsKN5V/cIll0b/tf9iQ5464nc+nM///U+UVxqT8ipeR7ThIPwuWX98cFZFQNNGkha4PaYap/osquEpRAJOcTqZf2K95ipeQ+5Hhw00mK0hcV1QT/7maTUqCHDfBCaD+uYAFvaNBXOYpdoIGM9cMk7Qjc/yowLDm+DpmJek54MWmN+iZ0YtDEhMSh//QPFMLPT5Ucat+qRTen1HZNGdxfZ7NIIDL3dNKVDN+vDUbW7rjvPyxA8Rtj4JplI9ENqpzRq4m1sDWUTk2hJYw9Ec1kGo7AFKRmOS6DRbwUn5Ptdc";
```
Comment (msbw2) [inline test/Microsoft.IdentityModel.Protocols.WsFederation.Tests/WsFederationMessageTests.cs]: Does this cert expire?

File: test/Microsoft.IdentityModel.TestUtils/KeyingMaterial.cs [hunk 9]
```diff
@@ -30,6 +33,7 @@ static public class KeyingMaterial
...
         public static string SelfSigned1024_SHA256_Public = "MIICWTCCAcKgAwIBAgIQPq8RWqoOL51G/qUxq7tBizANBgkqhkiG9w0BAQsFADA1MTMwMQYDVQQDHioAUwBlAGwAZgBTAGkAZwBuAGUAZAAxADAAMgA0AF8AUwBIAEEAMgA1ADYwHhcNMTQxMjI2MTUyNzAwWhcNMzkxMjMxMjM1OTU5WjA1MTMwMQYDVQQDHioAUwBlAGwAZgBTAGkAZwBuAGUAZAAxADAAMgA0AF8AUwBIAEEAMgA1ADYwgZ8wDQYJKoZIhvcNAQEBBQADgY0AMIGJAoGBAMl0ubbO3e3FpeMN2/gjz0YElurF0UCT25777CmNfzpvxtuvziUxa/ALMxmz+sVLzKb6kMuLNagU12cYZS3ksgntqxz8R/VyZO33hysNAMeMcdeOMhMm6ubTXEqqdHAb4TUPhme6o7Bvpcx3yPlxECoy6XE5A0VlYBgWzqMJN6F5AgMBAAGjajBoMGYGA1UdAQRfMF2AEKiIeaB2ShCC/Mrhyxg2Lf+hNzA1MTMwMQYDVQQDHioAUwBlAGwAZgBTAGkAZwBuAGUAZAAxADAAMgA0AF8AUwBIAEEAMgA1ADaCED6vEVqqDi+dRv6lMau7QYswDQYJKoZIhvcNAQELBQADgYEAaqChtfN/l6xTcMItwFG9jhDPuWeLDXAplM0vSwbia1fIaAXdcFRSaH+5QwqoQSDROcfiWRbPNWhFXfzOj7FEBmtbGifiqDvHislRHYrqnz9FRKiay0KYn0tJ2RUsTlKxZNz0WVu9M05wJjYH4TB04ad5FhgxJZ2h/y1X+An4a/o=";
+#pragma warning disable SYSLIB0057 // X509CertificateLoader does not have the correct overloads for this constructor
```
Comment (msbw2) [inline test/Microsoft.IdentityModel.TestUtils/KeyingMaterial.cs]: Is there a separate issue to track this instance?

File: test/Microsoft.IdentityModel.TestUtils/KeyingMaterial.cs [hunk 10]
```diff
@@ -97,11 +100,11 @@ static public class KeyingMaterial
         public static X509SecurityKey DefaultX509Key_2048_With_KeyId = new X509SecurityKey(DefaultCert_2048) { KeyId = DefaultX509Key_2048_KeyId };
         public static SigningCredentials DefaultX509SigningCreds_2048_RsaSha2_Sha2 = new SigningCredentials(DefaultX509Key_2048, SecurityAlgorithms.RsaSha256Signature, SecurityAlgorithms.Sha256Digest);
         public static X509Certificate2 DefaultAsymmetricCert_2048 = new X509Certificate2(Convert.FromBase64String(DefaultX509Data_2048), CertPassword, X509KeyStorageFlags.MachineKeySet);
-
+#pragma warning restore SYSLIB0057
```
Comment (msbw2) [inline test/Microsoft.IdentityModel.TestUtils/KeyingMaterial.cs]: This section with `SYSLIB0057` seems to cover too much. Can you move this to only encompass the necessary parts and add more pragmas as needed with some tracking items for them if none already exist?

File: test/Microsoft.IdentityModel.TestUtils/CertificateHelper.cs [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+// Licensed under the MIT License.
+
+using System.Security.Cryptography.X509Certificates;
+
+namespace Microsoft.IdentityModel.TestUtils
+{
+    /// <summary>
+    /// Helper class to load X509Certificate2 from byte array.
+    /// </summary>
+    public class CertificateHelper
```
Comment (pmaytak) [inline test/Microsoft.IdentityModel.TestUtils/CertificateHelper.cs:12]: I think it's okay to share internals with our test libraries, since tests reference the projects as source, so this class may not be needed.

**Review by pmaytak — APPROVED** [review]

File: test/Microsoft.IdentityModel.Protocols.WsFederation.Tests/WsFederationMessageTests.cs (hunk 8 above)
Comment (kllysng) [reply test/Microsoft.IdentityModel.Protocols.WsFederation.Tests/WsFederationMessageTests.cs]: I don't know but looking at the blame, these tests have been here for years. I am guessing no.

File: test/Microsoft.IdentityModel.Protocols.WsFederation.Tests/WsFederationMessageTests.cs (hunk 7 above)
Comment (kllysng) [reply test/Microsoft.IdentityModel.Protocols.WsFederation.Tests/WsFederationMessageTests.cs]: There were some intermediate changes with huge diffs, I must have missed this. Will fix, thank you!

File: test/Microsoft.IdentityModel.TestUtils/CertificateHelper.cs (hunk 11 above)
Comment (kllysng) [reply test/Microsoft.IdentityModel.TestUtils/CertificateHelper.cs:12]: I'm going to keep the test public helper class as there were issues with sharing internals with one of the test projects.

File: src/Microsoft.IdentityModel.Tokens/X509CertificateHelper.cs (hunk 6 above)
Comment (kllysng) [reply src/Microsoft.IdentityModel.Tokens/X509CertificateHelper.cs]: Talked offline and looks like we prefer to use it in net8 - updating.

File: test/Microsoft.IdentityModel.TestUtils/KeyingMaterial.cs (hunk 9 above)
Comment (kllysng) [reply test/Microsoft.IdentityModel.TestUtils/KeyingMaterial.cs]: Do you mean an issue opened on dotnet or an internal one for us to monitor whether they've added the correct overloads?

File: src/Microsoft.IdentityModel.Tokens/X509CertificateHelper.cs (hunk 6 above)
Comment (kllysng) [reply src/Microsoft.IdentityModel.Tokens/X509CertificateHelper.cs]: Update: image

Comment (kllysng) [reply src/Microsoft.IdentityModel.Tokens/X509CertificateHelper.cs]: @pmaytak found that: 

For .NET 9, in the Definition, the assembly is SystemSecurity.Cryptography.dll.
[X509CertificateLoader.LoadCertificate Method (System.Security.Cryptography.X509Certificates) | Microsoft Learn](https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.x509certificateloader.loadcertificate?view=net-9.0#definition)
 
For .NET 8, it's the backwards compat package: Microsoft.Bcl.Cryptography.dll
[X509CertificateLoader.LoadCertificate Method (System.Security.Cryptography.X509Certificates) | Microsoft Learn](https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.x509certificateloader.loadcertificate?view=net-8.0#definition)

I would rather not introduce a new dependency to also use this api in .net 8. Thoughts?

File: test/Microsoft.IdentityModel.TestUtils/KeyingMaterial.cs (hunk 10 above)
Comment (kllysng) [reply test/Microsoft.IdentityModel.TestUtils/KeyingMaterial.cs]: Introducing a helper with the warning disable encompassing only the X509Certificate2 constructor

File: test/Microsoft.IdentityModel.TestUtils/KeyingMaterial.cs (hunk 9 above)
Comment (kllysng) [reply test/Microsoft.IdentityModel.TestUtils/KeyingMaterial.cs]: I opened an [issue ](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/2833)and linked it.

File: src/Microsoft.IdentityModel.Tokens/X509CertificateHelper.cs (hunk 6 above)
Comment (kllysng) [reply src/Microsoft.IdentityModel.Tokens/X509CertificateHelper.cs]: Discussed offline and we don't want to add another dependency to use this in .net 8.

File: src/Microsoft.IdentityModel.Tokens/CertificateHelper.cs [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+using System.Security.Cryptography.X509Certificates;
+
+namespace Microsoft.IdentityModel.Tokens
+{
+    /// <summary>
+    /// Helper class to load X509Certificate2 from byte array.
+    /// </summary>
+    internal class CertificateHelper
+    {
+        public static X509Certificate2 LoadX509Certificate(byte[] data)
```
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Tokens/CertificateHelper.cs:13]: If you wanted to, you could add another overload that takes a string in base64format, then convert in the method.

Reducing code for the caller.

**Review by brentschmaltz — APPROVED** [review]
Small suggestion for an additional API (optional)

File: src/Microsoft.IdentityModel.Tokens/CertificateHelper.cs (hunk 12 above)
Comment (kllysng) [reply src/Microsoft.IdentityModel.Tokens/CertificateHelper.cs:13]: I promise I will do this in a follow up PR, I don't want to reset any approvals at this stage 🙏 🙇

**Review by msbw2 — APPROVED** [review]

File: src/Microsoft.IdentityModel.Tokens/CertificateHelper.cs (hunk 12 above)
Comment (kllysng) [reply src/Microsoft.IdentityModel.Tokens/CertificateHelper.cs:13]: As promised: #2834
